File: typescript-client-sdk/src/sdk/models/operations/searchcrmnotes.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class SearchCrmNotesQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=allFields" })
  allFields?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=cursor" })
  cursor?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=limit" })
  limit?: number;
}


export class SearchCrmNotesRequestBodyFilters extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=content" })
  content?: shared.StringFilter;

  @SpeakeasyMetadata({ data: "json, name=contentHtml" })
  contentHtml?: shared.StringFilter;

  @SpeakeasyMetadata({ data: "json, name=contentText" })
  contentText?: shared.StringFilter;

  @SpeakeasyMetadata({ data: "json, name=createdTime" })
  createdTime?: shared.DateFilter;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: shared.StringFilter;

  @SpeakeasyMetadata({ data: "json, name=modifiedTime" })
  modifiedTime?: shared.DateFilter;

  @SpeakeasyMetadata({ data: "json, name=nativeId" })
  nativeId?: shared.StringFilter;
}


export class SearchCrmNotesRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=filters" })
  filters?: SearchCrmNotesRequestBodyFilters;
}


export class SearchCrmNotesResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=nextPageCursor" })
  nextPageCursor?: string;

  @SpeakeasyMetadata({ data: "json, name=notes", elemType: shared.Note })
  notes?: shared.Note[];
}


export class SearchCrmNotesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: SearchCrmNotesQueryParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: SearchCrmNotesRequestBody;
}


export class SearchCrmNotesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: SearchCrmNotesResponseBody;
}

File: go-client-sdk/pkg/models/operations/searchcrmnotes.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type SearchCrmNotesQueryParams struct {
	AccessToken string   `queryParam:"style=form,explode=true,name=accessToken"`
	AllFields   *bool    `queryParam:"style=form,explode=true,name=allFields"`
	Cursor      *string  `queryParam:"style=form,explode=true,name=cursor"`
	Limit       *float64 `queryParam:"style=form,explode=true,name=limit"`
}

type SearchCrmNotesRequestBodyFilters struct {
	Content      *shared.StringFilter `json:"content,omitempty"`
	ContentHTML  *shared.StringFilter `json:"contentHtml,omitempty"`
	ContentText  *shared.StringFilter `json:"contentText,omitempty"`
	CreatedTime  *shared.DateFilter   `json:"createdTime,omitempty"`
	ID           *shared.StringFilter `json:"id,omitempty"`
	ModifiedTime *shared.DateFilter   `json:"modifiedTime,omitempty"`
	NativeID     *shared.StringFilter `json:"nativeId,omitempty"`
}

type SearchCrmNotesRequestBody struct {
	Filters *SearchCrmNotesRequestBodyFilters `json:"filters,omitempty"`
}

type SearchCrmNotesResponseBody struct {
	NextPageCursor *string       `json:"nextPageCursor,omitempty"`
	Notes          []shared.Note `json:"notes,omitempty"`
}

type SearchCrmNotesRequest struct {
	QueryParams SearchCrmNotesQueryParams
	Request     *SearchCrmNotesRequestBody `request:"mediaType=application/json"`
}

type SearchCrmNotesResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *SearchCrmNotesResponseBody
}
